src/test/java/com/google/devtools/build/lib/remote/SimpleBlobStoreFactoryTest.java

```diff
@@ -0,0 +1,190 @@
+    RemoteOptions options = Options.getDefaults(RemoteOptions.class);
+    options.remoteHttpCache = "http://doesnotesist.com";
+    options.remoteCacheProxy = "bad-proxy";
+    options.diskCache = null;
+
+    expectedEx.expect(Exception.class);
+    expectedEx.expectMessage("Remote cache proxy unsupported: bad-proxy");
+
+    SimpleBlobStoreFactory.create(options, /* creds */ null, workingDirectory);
+  }
+
+  @Test
+  public void createHttpCacheWithoutProxy() throws IOException {
+    RemoteOptions options = Options.getDefaults(RemoteOptions.class);
+    options.remoteHttpCache = "http://doesnotesist.com";
+    options.diskCache = null;
+
+    SimpleBlobStore blobStore = SimpleBlobStoreFactory.create(options, /* creds */ null, workingDirectory);
+
+    assertThat(blobStore).isNotNull();
+    assertThat(blobStore).isInstanceOf(HttpBlobStore.class);
+  }
+
+  @Test
+  public void createDiskCache() throws IOException {
+    RemoteOptions options = Options.getDefaults(RemoteOptions.class);
+    options.diskCache = PathFragment.create("/etc/something/cache/here");
+
+    when(workingDirectory.getRelative(options.diskCache)).thenReturn(relativeDirectory);
+    when(relativeDirectory.exists()).thenReturn(true);
+
+    SimpleBlobStore blobStore = SimpleBlobStoreFactory.create(options, /* creds */ null, workingDirectory);
+
+    assertThat(blobStore).isNotNull();
+    assertThat(blobStore).isInstanceOf(OnDiskBlobStore.class);
+  }
+
+  @Test
+  public void isRemoteCacheOptions_true_whenHttpCache() {
+    RemoteOptions options = new RemoteOptions();
+    options.remoteHttpCache = "http://doesnotexist:90";
+    options.diskCache = null;
+
+    assertThat(SimpleBlobStoreFactory.isRemoteCacheOptions(options)).isTrue();
+  }
+
+  @Test
+  public void isRemoteCacheOptions_true_whenDiskCache() {
+    RemoteOptions options = new RemoteOptions();
+    options.remoteHttpCache = null;
+    options.diskCache = PathFragment.create("/etc/something/cache/here");
+
+    assertThat(SimpleBlobStoreFactory.isRemoteCacheOptions(options)).isTrue();
+  }
+
+  @Test
+  public void isRemoteCacheOptions_true_whenHttpAndDiskCache() {
+    RemoteOptions options = new RemoteOptions();
+    options.remoteHttpCache = "http://doesnotexist:90";
+    options.diskCache = PathFragment.create("/etc/something/cache/here");
+
+    assertThat(SimpleBlobStoreFactory.isRemoteCacheOptions(options)).isTrue();
+  }
+
+  @Test
+  public void isRemoteCacheOptions_false_whenDiskCacheOptionEmpty() {
+    RemoteOptions options = new RemoteOptions();
+    options.diskCache = PathFragment.EMPTY_FRAGMENT;
+    options.remoteHttpCache = null;
+
+    assertThat(SimpleBlobStoreFactory.isRemoteCacheOptions(options)).isFalse();
+  }
+
+  @Test
+  public void isRemoteCacheOptions_false_whenRemoteHttpCacheEmpty() {
+    RemoteOptions options = new RemoteOptions();
+    options.remoteHttpCache = "";
+
+    // todo #7650: flip to false
```

Comment:
TODO(ishikhman): Flip to false. See #7650

Reply:
done